### DISCUSSION TITLE
Initial Backup Fails without useful information

### DISCUSSION THREAD

**Discussion by frahmlan** [nextcloud/all-in-one#4390] [Questions] [answered]
On a fresh AIO install, the initial Backup fails without useful information.

Backup container logs
`database-dump is missing. Cannot perform backup!
Please check the database container logs!
`

Database Container Logs
`Setting postgres values...
chmod: /var/run/postgresql: Operation not permitted

PostgreSQL Database directory appears to contain a database; Skipping initialization

2024-03-14 09:20:41.983 CDT [14] LOG:  starting PostgreSQL 15.6 on x86_64-pc-linux-musl, compiled by gcc (Alpine 13.2.1_git20231014) 13.2.1 20231014, 64-bit
2024-03-14 09:20:41.983 CDT [14] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2024-03-14 09:20:41.983 CDT [14] LOG:  listening on IPv6 address "::", port 5432
2024-03-14 09:20:41.985 CDT [14] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2024-03-14 09:20:41.993 CDT [24] LOG:  database system was interrupted; last known up at 2024-03-14 08:49:39 CDT
2024-03-14 09:20:42.400 CDT [24] LOG:  database system was not properly shut down; automatic recovery in progress
2024-03-14 09:20:42.404 CDT [24] LOG:  redo starts at 0/187DD2D0
2024-03-14 09:20:42.409 CDT [24] LOG:  invalid record length at 0/188CCC00: wanted 24, got 0
2024-03-14 09:20:42.409 CDT [24] LOG:  redo done at 0/188CCBC8 system usage: CPU: user: 0.00 s, system: 0.00 s, elapsed: 0.00 s
2024-03-14 09:20:42.458 CDT [14] LOG:  database system is ready to accept connections
`

**Comment by szaimen** [comment]
Hi, I fear I cannot reproduce your issue. Which docker version ans OS are you using?

**Reply by frahmlan** [reply]
Docker version 24.0.5, build 24.0.5-0ubuntu1
docker-compose version 1.29.2

Ubuntu 23.10

I've been starting my mastercontainer from docker-compose

**Reply by szaimen** [reply]
> Ubuntu 23.10

Right, I think I saw that before already. There seems to be a bug in docker and/or ubuntu 23.10 that does not report stop attempts correctly to the container so that the logic that triggers the database dump is not triggered.

Always best to stick with Ubuntu LTS releases for servers.

**Reply by frahmlan** [reply]
I'll try to reinstall Ubuntu LTS when I get a chance and see if that solves the issue. Thanks for the help

**Reply by rkl110** [reply]
Same with 24.04 LTS Ubuntu.

**Reply by kczack** [reply]
I have the same problem with 24.04 LTS Ubuntu.

**Reply by Pacos** [reply]
I had the same issue with Ubuntu 24.04 LTS. Switching to the [official Docker install](https://docs.docker.com/engine/install/ubuntu/#install-docker-ce) - not the one in the Ubuntu base packages - seems to have solved the issue.

**Comment by karbowiak** [comment]
Just discovered the same problem, also running Ubuntu 24.04 LTS..

But why it would be an issue limited to that i don't get, because everything should be handled inside the containers, with the host OS being irrelevant.

edit://
Checked the usual suspects and read the README, i noticed it talking about alpine and fuse - and lo and behold, 24.04 LTS doesn't come with fuse by default, so i installed it - i'll see what happens over the next day or so.
I also added the DAILY_BACKUP and such ENVs..

So the entire run command is now.
`docker run --init --sig-proxy=false --name nextcloud-aio-mastercontainer --restart always --env AUTOMATIC_UPDATES=1 --env DAILY_BACKUP=1 --env NEXTCLOUD_DATADIR=/containers/nextcloud --env AIO_COMMUNITY_CONTAINERS="facerecognition memories stalwart caddy vaultwarden fail2ban" --publish 80:80 --publish 8080:8080 --publish 8443:8443 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest`

**Reply by szaimen** [reply]
Unfortunately the OS is still the culprit here as the docker runtime does not send the stop signals correctly to the containers on some docker versions.

**Reply by karbowiak** [reply]
@szaimen i see, would it help to install the non-ubuntu-apt version of docker?
As in installing it directly from the Docker apt repo? https://docs.docker.com/engine/install/ubuntu/

**Reply by szaimen** [reply]
yes, installing directly from the docker apt repo should fix this

**Reply by karbowiak** [reply]
@szaimen alrighty, i've removed docker, and installed docker from the docker apt repo, i'll see what happens - initially clicking "Create Backup" in the AIO panel yields the same error, but i guess stuff needs to be running for a while before anything happens 😅 

I'll keep you informed ! 😄

**Reply by szaimen** [reply]
> stuff needs to be running for a while before anything happens 😅

Yes indeed. First start the containers again. Afterwards the backup should work.

**Reply by karbowiak** [reply, marked as answer]
Yessir, that did the trick!

So if using Ubuntu 24.04 LTS..

1. Install fuse `apt install fuse`
2. Install docker-ce from https://docs.docker.com/engine/install/ubuntu/

And then everything should work - coolio, thanks so much! 😄
